flink-autoscaler/src/main/java/org/apache/flink/autoscaler/realizer/ScalingRealizer.java

```diff
@@ -33,8 +33,13 @@
 @Experimental
 public interface ScalingRealizer<KEY, Context extends JobAutoScalerContext<KEY>> {
 
-    /** Update job's parallelism to parallelismOverrides. */
-    void realizeParallelismOverrides(Context context, Map<String, String> parallelismOverrides);
+    /**
+     * Update job's parallelism to parallelismOverrides.
+     *
+     * @throws Exception Error during realize parallelism overrides.
+     */
+    void realizeParallelismOverrides(Context context, Map<String, String> parallelismOverrides)
+            throws Exception;
 
     /** Updates the TaskManager memory configuration. */
     void realizeConfigOverrides(Context context, ConfigChanges configChanges);
```

Comment:
I'm thinking should these 2 methods keep same semantic?  Such as: assert ` throws Exception` together.

It's a little wired: one throws, one not.

WDYT?

Reply:
+1